packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -0,0 +1,607 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import 'layer.dart';
+import 'object.dart';
+
+/// The result after handling a [SelectionEvent].
+///
+/// [SelectionEvent]s are sent from [SelectionRegistrar] to be handled by
+/// [SelectionHandler.dispatchSelectionEvent]. The subclasses of
+/// [SelectionHandler] or [Selectable] must return appropriate
+/// [SelectionResult]s after handling the events.
+///
+/// This is used by the [SelectionContainer] to determine how a selection
+/// expands across its [Selectable] children.
+enum SelectionResult {
+  /// There is nothing left to select forward in this [Selectable], and further
+  /// selection should extend to the next [Selectable] in screen order.
+  ///
+  /// {@template flutter.rendering.selection.SelectionResult.footNote}
+  /// This is used after subclasses [SelectionHandler] or [Selectable] handled
+  /// [SelectionEdgeUpdateEvent].
+  /// {@endtemplate}
+  next,
+  /// Selection does not reach this [Selectable] and is located before it in
+  /// screen order.
+  ///
+  /// {@macro flutter.rendering.selection.SelectionResult.footNote}
+  previous,
+  /// Selection ends in this [Selectable].
+  ///
+  /// Part of the [Selectable] may or may not be selected, but there is still
+  /// content to select forward or backward.
+  ///
+  /// {@macro flutter.rendering.selection.SelectionResult.footNote}
+  end,
+  /// The result can't be determined in this frame.
+  ///
+  /// This is typically used when the subtree is scrolling to reveal more
+  /// content.
+  ///
+  /// {@macro flutter.rendering.selection.SelectionResult.footNote}
+  // See `_SelectableRegionState._triggerSelectionEndEdgeUpdate` for how this
+  // result affects the selection.
```

Comment:
Did you mean for this to be triple slash `///`?

Reply:
This is a comment for code reader, the customer that implements Selectable does not need to know how `pending` is used. They just need to know they should return pending if the selection can't be determine in this frame.

Reply:
Ah I see, sounds good.